### PR TITLE
[MINOR] Fix for BUILD FAILURE for Scala 2.11

### PR DESCRIPTION
**Pull request by jaceklaskowski**
It was introduced in 917d3fc069fb9ea1c1487119c9c12b373f4f9b77

/cc @cloud-fan @rxin

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala [hunk 1]
```diff
@@ -68,29 +68,12 @@ private[sql] class JSONRelation(
...
     (@transient val sqlContext: SQLContext)
   extends HadoopFsRelation(maybePartitionSpec, parameters) {
 
-  def this(
-      inputRDD: Option[RDD[String]],
-      maybeDataSchema: Option[StructType],
-      maybePartitionSpec: Option[PartitionSpec],
-      userDefinedPartitionColumns: Option[StructType],
-      paths: Array[String] = Array.empty[String],
-      parameters: Map[String, String] = Map.empty[String, String])(sqlContext: SQLContext) = {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala]: scala 2.11 doesn't support override curry constructor?

Comment (jaceklaskowski) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala]: The error message was:

```
[error] /Users/jacek/dev/oss/spark/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala:66:
in class JSONRelation, multiple overloaded alternatives of constructor
JSONRelation define default arguments.
[error] private[sql] class JSONRelation(
[error]                    ^
```

**Comment by srowen** [comment]
I think that's the right sort of fix, yes. I recall scala 2.11 is strict on this now. If it still compiles in 2.10 I think we're good to go

**Comment by rxin** [comment]
I've merged this. Thanks.

**Pull request closed without merging**